docs/src/main/asciidoc/spring-integration-pubsub.adoc

```diff
@@ -71,29 +71,32 @@ Then, we declare a `PubSubInboundChannelAdapter` bean.
 It requires the channel we just created and a `SubscriberFactory`, which creates `Subscriber` objects from the Google Cloud Java Client for Pub/Sub.
 The Spring Boot starter for GCP Pub/Sub provides a configured `PubSubSubscriberOperations` object.
 
-The `PubSubInboundChannelAdapter` supports three acknowledgement modes, with `AckMode.AUTO` being the default value;
-
-Automatic acking (`AckMode.AUTO`)
-
-A message is acked with GCP Pub/Sub if the adapter sent it to the channel and no exceptions were thrown.
-
-If a `RuntimeException` is thrown while the message is processed, then the message is handled in the following way:
-
-* If an error channel is configured for the adapter (`pubSubInboundChannelAdapater.getErrorChannel() != null`) then the message will be acked and forwarded with the error to the error channel.
-* If no error channel is configured for the adapter, then the message is nacked.
-
-Automatic acking OK (`AckMode.AUTO_ACK`)
-
-A message is acked with GCP Pub/Sub if the adapter sent it to the channel and no exceptions were thrown.
-If a `RuntimeException` is thrown while the message is processed, then the message is neither acked / nor nacked.
-
-This is useful when using the subscription's ack deadline timeout as a retry delivery backoff mechanism.
-
-Manually acking (`AckMode.MANUAL`)
-
+===== Acknowledging messages and handling failures
+When working with Cloud Pub/Sub, it is important to understand the concept of `ackDeadline` -- the amount of time Cloud Pub/Sub will wait until attempting redelivery of an outstanding message.
+While acknowledging (acking) a message removes it from Pub/Sub's known outstanding messages, nacking a message simply means resetting its acknowledgement deadline to 0, forcing immediate redelivery.
+A global `ackDeadline` exists on each subscription. Additionally, the Cloud Pub/Sub client library can extend each streamed message's `ackDeadline` up to `spring.cloud.gcp.pubsub.subscriber.max-ack-extension-period` seconds (in `spring-cloud-gcp`, this is set to 0, which prevents automatic deadline extension, but in a standalone client library, the default is 60 minutes).
+
+The `PubSubInboundChannelAdapter` supports three acknowledgement modes, with `AckMode.AUTO` being the default value:
+
+* Automatic acking and nacking (`AckMode.AUTO`) +
+If the message was successfully processed with no exceptions, an acknowledgement is automatically sent to Cloud Pub/Sub.
+In case of processing failure, a nack (request for immediate redelivery) is automatically sent to Cloud Pub/Sub.
+
+* Automatic acking only (`AckMode.AUTO_ACK`) +
+If the message was successfully processed with no exceptions, an acknowledgement is automatically sent to Cloud Pub/Sub.
+In case of processing failure, the adapter neither acks nor nacks the message, resulting in the same behavior as `MANUAL` mode.
```

Comment:
```suggestion
In case of a processing failure, the adapter neither acks nor nacks the message, resulting in the same behavior as `MANUAL` mode, where the the application logic can decide to nack manually or allow for eventual redelivery.
```